plutus-contract/src/Plutus/Contract/StateMachine/OnChain.hs

```diff
@@ -92,20 +93,20 @@ machineAddress = scriptAddress . validatorInstance
 -- | Turn a state machine into a validator script.
 mkValidator :: forall s i. (PlutusTx.IsData s) => StateMachine s i -> ValidatorType (StateMachine s i)
 mkValidator (StateMachine step isFinal check) currentState input ptx =
-    let vl = txInInfoValue (findOwnInput ptx)
+    let vl = maybe (error ()) (txOutValue . txInInfoResolved) (findOwnInput ptx)
```

Comment:
Hmm, this is vaguely unsatisfactory, but not much we can do about it. We're exposing the error case from `findOwnInput` to users, which I guess is okay.

Reply:
Yeah I pushed it out to here, previously `findOwnInput` just failed.